tools/LogCollector/source/omiHighCPUDiagnostics.sh

```diff
@@ -0,0 +1,207 @@
+#!/bin/bash
+#set -x
+declare -a ppuc
+declare -a pt
+cpu_threshold=20.0
+rescan_interval_sec=5
+default_runtime_min=1440
+count=`expr $default_runtime_min / $rescan_interval_sec`
+omiagent_trace_path=./omiagent_trace
+omiagent_trace_bak_path=./omiagent_trace.bak
+omi_log_path="/var/opt/omi/log"
+scx_log_path="/var/opt/microsoft/scx/log"
+omi_log_file="${omi_log_path}/omiserver.log"
+omiagent_log_file="${omi_log_path}/omiagent.root.root.log"
+scx_log_file="${scx_log_path}/scx.log"
+
+if [ "$1" = "--help" ]; then
+    echo "This scipt runs for $default_runtime_min minutes, and collect info about omiagent process with high cpu utilization ( > $cpu_threshold% )."
+    echo "When CPU utilization is high it takes the stack trace snapshot for the omiagent for 10 times in interval of ${rescan_interval_sec} sec."
+    echo "Then rescan the system in interval of ${rescan_interval_sec} sec."
+    echo "It also collects the diff of scx and omi logs in every ${rescan_interval_sec} sec"
+    echo "  Option            Description"
+    echo "--cpu-threshold     set cpu threashold, default $cpu_threshold%"
+    echo "--runtime-in-min    set run time in minutes, default $default_runtime_min min <=> `expr $default_runtime_min / 60` hours"
```

Comment:
not sure it is working properly... put in 5 min but was > 20 min

Reply:
the timing will be affected if the script is collecting traces all the time.